stl/inc/xutility

```diff
@@ -744,17 +744,20 @@ concept input_or_output_iterator = requires(_It __i) {
 
 // CONCEPT sentinel_for
 template <class _Se, class _It>
-concept sentinel_for = semiregular<_Se> && input_or_output_iterator<_It> && _Weakly_equality_comparable_with<_Se, _It>;
+concept sentinel_for = semiregular<_Se>
+    && input_or_output_iterator<_It>
+    && _Weakly_equality_comparable_with<_Se, _It>;
 // clang-format on
 
-// VARIABLE TEMPLATE disable_sized_sentinel
+// VARIABLE TEMPLATE disable_sized_sentinel_for
 template <class _Se, class _It>
-inline constexpr bool disable_sized_sentinel = false;
+inline constexpr bool disable_sized_sentinel_for = false;
 
 // CONCEPT sized_sentinel_for
 // clang-format off
 template <class _Se, class _It>
-concept sized_sentinel_for = sentinel_for<_Se, _It> && !disable_sized_sentinel<remove_cv_t<_Se>, remove_cv_t<_It>>
+concept sized_sentinel_for = sentinel_for<_Se, _It>
+    && !disable_sized_sentinel_for<remove_cv_t<_Se>, remove_cv_t<_It>>

```

Comment:
Wasnt there a paper that argued about `enabled_sized_sentinel_for` or generally turn all `disabled_foo` into their `enabled_foo` opposite

Reply:
Yes. Ironically enough, that paper was [P1871R0 "Should concepts be enabled or disabled?"](https://wg21.link/p1871r0). Discussion of that proposal in LEWG apparently validated the original design, which is:

* Passing an argument to a library component that constrains that argument is an implicit claim that the argument either doesn't satisfy the constraint (doesn't meet the syntactic requirements) or that it does model the constraint (meets the syntax and the semantic requirements) so opt-in/opt-out are not generally necessary.

* opt-in/opt-out facilities are only provided when we want to enable the library to assume that an argument that models some concept `X` additionally models some refinement `Y` of `X` when it satisfies `Y`. (I've been calling this "semantic promotion", but it's not a well-established term.) The choice of opt-in vs. opt-out depends on the expected frequency of false positives. 

Examples:
1. Library components that accept `range` arguments can promote to `sized_range`, i.e., assume a `range` argument `R` models `sized_range` when `sized_range<R>` is `true`. Arguments that model `range` and additionally satisfy but do not model `sized_range` are rare, but we don't want to ban them from the standard library altogether, so we provide a `disable_sized_range` opt-out mechanism.
2. There is little or no syntactic distinction between `range` and `view` concepts - we know there are lots of semiregular `range`s that nonetheless do not model `view` - but view composition wants to promote `range`s to `view`s when possible. The `view` concept is (now) purely opt-in (a type either derives from `ranges::view_base` or specializes `ranges::enable_view`) to avoid false positives.

So, once we've decided whether a concept should be opt-in vs. opt-out, the "disable" vs. "enable" choice is fairly clear: I want to use `enable` names to opt in, and `disable` names to opt out. It's less intuitive to specialize `enable_sized_range` to `false` to opt out of `sized_range` than it is to specialize `disable_sized_range` to `true`: the "default" is already `true`, and I want to disable it.

The proposal also mentioned in passing that it was weird that we had renamed `sized_sentinel` to `sized_sentinel_for` without renaming `disable_sized_sentinel` to `disable_sized_sentinel_for`, and LEWG decided to keep just that part.

Reply:
Thanks for the detailed explanation 👍